### PR TITLE
Create a program that works with a file

### PR DESCRIPTION
**Pull request by VLONE300**
image

### CODE REVIEW

File: home_works/lesson14/app.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from home_works.lesson14.file_worker import FileWorker
+
+
+def app():
+    fw = FileWorker('E:\\Python\\TMS_py94\\qwerty.json')
```
Comment (nicilytop-tms) [inline home_works/lesson14/app.py]: 🫣 is it windows?..

File: home_works/lesson14/file_worker.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from home_works.lesson14.handlers import JsonHandler, TxtHandler
+import os
+
+
+class FileWorker:
+    def __init__(self, path):
+        self.path = path
+        self.handler = None
+        if self.get_file_extension() == '.json':
```
Comment (nicilytop-tms) [inline home_works/lesson14/file_worker.py]: You can just set class dict like:
```Python
extensions_handler = {'.json': JsonHandler}
```
And here do some `self.handler = extensions_handler[self.get_file_extension()]`

File: home_works/lesson14/file_worker.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            self.handler = JsonHandler(self.path)
+        elif self.get_file_extension() == '.txt':
+            self.handler = TxtHandler(self.path)
+
+    def get_file_extension(self):
+        file_name, file_extension = os.path.splitext(self.path)
```
Comment (nicilytop-tms) [inline home_works/lesson14/file_worker.py:13]: Nice.. using os I mean

**Review by nicilytop-tms — APPROVED** [review]
Nice

File: home_works/lesson14/handlers.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        return content
+
+    def append(self, data):
+        try:
+            with open(self.path) as file:
+                self._file = json.load(file)
+        except (json.JSONDecodeError, ValueError):
+            self._file = []
+
+        self._file.append(data)
```
Comment (nicilytop-tms) [inline home_works/lesson14/handlers.py:32]: hmm .. I think you should check that our data - is list

Comment (VLONE300) [reply home_works/lesson14/handlers.py:32]: I don't quite understand the conditions. If not the list, throw an error ?

**Pull request merged**